web/yo/app/scripts/directives/addmodifytherapy.js

```diff
@@ -0,0 +1,181 @@
+    .directive('addModifyTherapy', function ($rootScope, DatabaseConnector, _, $q, FirebaseModel, mainUtils, $window) {
+        return {
+            templateUrl: 'views/addModifyTherapy.html',
+            restrict: 'AE',
+            scope: false,
+            priority: -1,
+            controller: function ($scope) {
+                var therapyUuid = [];
+                if ($scope.modifyMode === true) {
+                    $scope.therapyUuid = [];
+                    $scope.therapy = [];
+                    $scope.addTherapyError = false;
+                    $scope.therapyResult = "";
+                    $scope.noData = true;
+                    initTherapy();
+                } else {
+                    clearData();
+                }
+
+                function clearData() {
+                    $scope.therapy = [{tags: []}];
+                    therapyUuid = [[]];
+                    $scope.addTherapyError = false;
+                    $scope.noData = true;
+                    $scope.therapyResult = "";
+                }
+
+                function refreshResult() {
+                    $scope.therapyResult = _.map($scope.therapy, function (element) {
+                        return (_.map(element.tags, function (name) {
+                            return name.drugName;
+                        }).join(" + "));
+                    }).join(", ");
+                }
+
+                function initTherapy() {
+                    var newTherapy = [];
+                    newTherapy = mainUtils.therapyStrToArr($scope.treatmentRef.name);
+                    firebase.database().ref('Drugs').once('value').then(function (snapshot) {
+                        var druglist = [];
+                        druglist = snapshot.val();
+                        for (var i = 0; i < newTherapy.length; i++) {
+                            $scope.therapy.push({tags: []});
+                            therapyUuid.push([]);
+                            var tem = [];
+                            var temuuid = [];
+                            for (var j = 0; j < newTherapy[i].length; j++) {
+                                tem.push(druglist[newTherapy[i][j]]);
+                                temuuid.push(druglist[newTherapy[i][j]].uuid);
+                            }
+                            $scope.therapy[i] = {tags: tem};
+                            therapyUuid[i] = temuuid;
+                        }
+                        $scope.therapy.push({tags: []});
+                        therapyUuid.push([]);
+                        refreshResult();
+                    }, function (error) {
+                        console.error((error));
+                    });
+                }
+
+                $scope.loadDrugs = function ($query) {
+                    return firebase.database().ref('Drugs').once('value').then(function (snapshot) {
+                        var drugs = [];
+                        $scope.addTherapyError = false;
```

Comment:
Why do we need to update `addTherapyError ` here ?

Reply:
When the user chose the same elements, the addTherapyError would show up. The addTherapyError would be hidden when the user activated one of the lines. And if the user chose same elements again, the addTherapyError would show again. Maybe more responsive?